### PR TITLE
Bug fix, timestamp fix, documentation update 

### PR DESCRIPTION
**Pull request by roycamp**
Fixed serverIndex Null Pointer bug
Preserve flume timestamp if it exists
Updated documentation

### CODE REVIEW

File: src/java/org/apache/cassandra/plugins/CassandraClient.java [hunk 1]
```diff
@@ -165,7 +163,7 @@ String get() throws NoServersAvailableException {
       if(this.servers.isEmpty()) {
         throw new NoServersAvailableException();
       }
-      return this.servers.get(this.serverIndex++);
+      return this.servers.get(0);
```
Comment (thobbs) [inline src/java/org/apache/cassandra/plugins/CassandraClient.java:166]: So, just incrementing serverIndex without making sure it hasn't gone off the end of the list was definitely broken, but we do want to cycle through the servers in our list.  If we always use 0, that won't happen.

I can make the change, but we probably just need to modulus by serverList.size().

File: src/java/org/apache/cassandra/plugins/SimpleCassandraSink.java [hunk 2]
```diff
@@ -63,7 +63,10 @@ public void open() throws IOException {
   @Override
   public void append(Event event) throws IOException, InterruptedException {
 
-    long timestamp = System.currentTimeMillis() * MILLI_TO_MICRO;
+    // Preserve timestamp from when the event was generated 
+    long timestamp = event.getTimestamp();
```
Comment (thobbs) [inline src/java/org/apache/cassandra/plugins/SimpleCassandraSink.java:63]: The main problem with doing this is that the TimeUUID we generate just below this won't use the same timestamp, so columns in the index rows won't necessarily match up with the time range in the row key.

It's possible to create a TimeUUID given a timestamp, but we would need to swap out UUID libraries and use something like this: https://github.com/rantav/hector/blob/master/core/src/main/java/me/prettyprint/cassandra/utils/TimeUUIDUtils.java

File: src/java/org/apache/cassandra/plugins/CassandraClient.java (hunk 1 above)
Comment (roycamp) [inline src/java/org/apache/cassandra/plugins/CassandraClient.java:166]: My understanding is that ServerSet.get() is only called from CassandraClient.open().  CassandraClient.open() is only called during start up and  when a Cassandra exception is thrown which results in ServerSet.markDead().  markDead() removes the server from the servers array thus causing the remaining servers to shift up in the array.  When the markDead() timeout expires, the server is added onto the end of the servers array.  This is how the cycling of servers is done.  Is there a scenario where open() is called which I am missing?

File: src/java/org/apache/cassandra/plugins/SimpleCassandraSink.java (hunk 2 above)
Comment (roycamp) [inline src/java/org/apache/cassandra/plugins/SimpleCassandraSink.java:63]: Ah, good catch.  I only accounted for the row key not the column TimeUUID.  I will commit a fix.

File: src/java/org/apache/cassandra/plugins/CassandraClient.java (hunk 1 above)
Comment (thobbs) [inline src/java/org/apache/cassandra/plugins/CassandraClient.java:166]: Ah, you're totally correct.  Thanks.

**Comment by roycamp** [comment]
I merged the most recent updates (maven support) -- hopefully did this right.
Added TimeUUID generation support for column names based on the event timestamp.

**Comment by roycamp** [comment]
Do I need to include the UUID jar (from http://johannburkard.de/software/uuid/) in the lib folder or will the maven dependency handle it?

**Comment by thobbs** [comment]
If the build were set up to make a single uber-jar, then we could forgo adding it to lib (and remove the requirement for the user to add multiple jars to the classpath).  I'll merge this in and then see if I can make that happen.

Thanks!

**Pull request merged**